src/AuthenticationRoute.ts

```diff
@@ -12,10 +12,26 @@ type FailureObject = {
   type?: string
 }
 
+const getSessionData = (request: FastifyRequest, key: string) => {
```

Comment:
I think it would make sense to add new functions to the session managers that know how to do this stuff instead of adding other wrapper functions out here. That way, to add a third supported library, folks just have to implement the session manager interface instead of editing these wrappers also. We could also then accept a session manager as a plugin option and there'd be no hardcoded bits that might need to be changed.

Reply:
I agree

Reply:
Makes sense. I have opened a [PR ](https://github.com/fastify/session/pull/50) on `@fastify/session` to add the new functions.
Correct me if I am wrong, once we have those function we wouldn't need any changes here in `fastify-passport`.  I will only need to update the documentation and add a reusable suite of tests which can be invoked with different session plugins.

Reply:
I think he means the interface inside this plugin.

```js
class SessionManager {
  getSession(session: any, key: string) {
    return session[key];
  }
  
  setSession(session: any, key: string, value: any) {
    session[key] = value;
  }
}

class SecureSessionManager {
  getSession(session: any, key: string) {
    return session.get(key);
  }
  
  setSession(session: any, key: string, value: any) {
    session.set(key, value)
  }
}
```

Reply:
> I think he means the interface inside this plugin.
> 
> ```js
> class SessionManager {
>   getSession(session: any, key: string) {
>     return session[key];
>   }
>   
>   setSession(session: any, key: string, value: any) {
>     session[key] = value;
>   }
> }
> 
> class SecureSessionManager {
>   getSession(session: any, key: string) {
>     return session.get(key);
>   }
>   
>   setSession(session: any, key: string, value: any) {
>     session.set(key, value)
>   }
> }
> ```

@climba03003  yeah probably he means inside the plugin but then that would add get/set methods to the `request.passport.sessionManager` object instead of `request.session`.  Adding those methods on @fastify/session would make it available on `request.session`.